lib/oli_web/live/workspaces/student.ex

```diff
@@ -24,21 +34,17 @@ defmodule OliWeb.Workspaces.Student do
   @impl Phoenix.LiveView
   def mount(params, _session, %{assigns: %{current_user: current_user}} = socket)
       when not is_nil(current_user) do
-    all_sections =
-      Sections.list_user_open_and_free_sections(current_user)
-      |> add_user_role(current_user)
-
     sections =
-      all_sections
-      |> filter_by_role(:student)
+      current_user.id
+      |> sections_where_user_is_student()
       |> add_instructors()
```

Comment:
**add_instructors part 1:** 1 query in total
```sql
SELECT e0.*, s1.id FROM enrollments AS e0
INNER JOIN sections AS s1 ON e0.section_id = s1.id
INNER JOIN enrollments_context_roles AS e2 ON e0.id = e2.enrollment_id
WHERE (
  (s1.id IN ($1) AND (e0.status = 'enrolled')) AND
  (e2.context_role_id = $2)
)

```
**add_instructors part 2:** 1 query per each instructor of section
```sql
SELECT * FROM users AS WHERE (id = $1)
```

Reply:
This was previously 1 query + 4 queries per section